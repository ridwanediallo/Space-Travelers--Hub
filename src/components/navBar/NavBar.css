.logo-img {
  width: 4rem;
  height: 4rem;
}

.link {
  text-decoration: none;
}

.link:active {
  color: #95bcf0;
  padding-bottom: 0.25rem;
  border-bottom: 3px solid #95bcf0;
}

.navbar-toggler-icon {
  background-color: black;
  filter: invert(100%);
}
