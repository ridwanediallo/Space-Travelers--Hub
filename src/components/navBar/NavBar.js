import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css';
import Logo from './planet.png';
import { Navbar, Container, Nav } from 'react-bootstrap';

const NavBar = () => {
  const activeLink = ({ isActive }) => {
    return {
      color: isActive ? '#95bcf0' : '',
      paddingBottom: isActive ? '0.25rem' : '',
      borderBottom: isActive ? '3px solid #95bcf0' : '',
    };
  };

  return (
    <Navbar collapseOnSelect expand="lg" bg="light" variant="dark">
      <Container>
        <Navbar.Brand href="#home" className="d-flex">
          <img className="logo" src={Logo} alt="logo" />
          <h2 className="ps-3 text-dark">Space Travelers Hub</h2>
        </Navbar.Brand>
        <Navbar.Toggle
          aria-controls="responsive-navbar-nav"
          className="text-dark"
        />
        <Navbar.Collapse id="responsive-navbar-nav">
          <Nav className="ms-auto">
            <Nav.Link href="#rockets">
              <NavLink
                style={activeLink}
                to="/"
                className="text-primary link"
                activeStyle="active"
              >
                Rockets
              </NavLink>
            </Nav.Link>
            <Nav.Link href="#missions">
              <NavLink
                style={activeLink}
                to="/missions"
                className="text-primary link"
                activeStyle="active"
              >
                Missions
              </NavLink>
            </Nav.Link>
            <Nav.Link href="#profile">
              <NavLink
                style={activeLink}
                to="/profile"
                className="text-primary link"
                activeStyle="active"
              >
                My Profile
              </NavLink>
            </Nav.Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default NavBar;
