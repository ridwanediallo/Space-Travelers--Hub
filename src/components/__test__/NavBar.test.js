import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../navBar/NavBar'

describe('Testing navBar component', () => {});
it('should render correctly NavBar component', () => {
  const tree = render(
    <Router>
      <NavBar />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});
