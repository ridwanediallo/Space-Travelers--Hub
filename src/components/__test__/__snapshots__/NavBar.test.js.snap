// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly NavBar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="border-bottom navbar navbar-expand-lg navbar-dark bg-white"
      >
        <div
          class="container"
        >
          <a
            class="d-flex navbar-brand"
            href="#home"
          >
            <div
              class="logo d-flex align-items-center py-2"
            >
              <img
                alt="logo"
                class="logo-img"
                src="planet.png"
              />
              <h2
                class="ps-3 text-dark"
              >
                Space Travelers Hub
              </h2>
            </div>
          </a>
          <button
            aria-controls="responsive-navbar-nav"
            aria-label="Toggle navigation"
            class="text-dark navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="responsive-navbar-nav"
          >
            <div
              class="ms-auto navbar-nav"
            >
              <a
                class="nav-link"
                data-rr-ui-event-key="#rockets"
                href="#rockets"
              >
                <a
                  aria-current="page"
                  class="text-primary link active"
                  href="/"
                  style="color: rgb(13, 110, 253); padding-bottom: 0.30rem; border-bottom: 3px solid #0d6efd;"
                >
                  Rockets
                </a>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#missions"
                href="#missions"
              >
                <a
                  class="text-primary link"
                  href="/missions"
                >
                  Missions
                </a>
              </a>
              <a
                class="nav-link"
                data-rr-ui-event-key="#profile"
                href="#profile"
              >
                <a
                  class="text-primary link"
                  href="/profile"
                >
                  My Profile
                </a>
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="border-bottom navbar navbar-expand-lg navbar-dark bg-white"
    >
      <div
        class="container"
      >
        <a
          class="d-flex navbar-brand"
          href="#home"
        >
          <div
            class="logo d-flex align-items-center py-2"
          >
            <img
              alt="logo"
              class="logo-img"
              src="planet.png"
            />
            <h2
              class="ps-3 text-dark"
            >
              Space Travelers Hub
            </h2>
          </div>
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="text-dark navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="ms-auto navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="#rockets"
              href="#rockets"
            >
              <a
                aria-current="page"
                class="text-primary link active"
                href="/"
                style="color: rgb(13, 110, 253); padding-bottom: 0.30rem; border-bottom: 3px solid #0d6efd;"
              >
                Rockets
              </a>
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="#missions"
              href="#missions"
            >
              <a
                class="text-primary link"
                href="/missions"
              >
                Missions
              </a>
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="#profile"
              href="#profile"
            >
              <a
                class="text-primary link"
                href="/profile"
              >
                My Profile
              </a>
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
